torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1803,6 +1824,18 @@ def _make_data_dependent_error(self, expr, unhinted_expr):
             # problem
         )
 
+    def _set_replacement(self, a: "sympy.Symbol", expr: "sympy.Expr") -> None:
+        """
+        Adds or updates a replacement for a symbol.
+        Use this instead of `self.replacements[a] = expr`.
+        """
+        if a not in self.replacements or expr != self.replacements[a]:
+            if torch._dynamo.config.print_specializations and isinstance(expr, (sympy.Integer, sympy.Float)):
+                # specializing to a constant, which is likely unexpected
+                torch._dynamo.guards.log.warning(f"Specializing {self.var_to_sources[a][0].name()} to {expr}")
```

Comment:
Call do torch._dynamo.guards log is not conventional; we're supposed to create a logger for this module like

```
log = logging.getLogger(__name__)
```

in fact, we already have one here.

Reply:
I didn't see anything on the command line when I used the symbolic shapes log. Is it disabled?

Reply:
I think the problem is that torch.fx.experimental is not in the same module namespace as torch._dynamo and torch._inductor, and the "PT2" logging only enables logs on certain parts of the module hierarchy. So the fix would be to enable it on torch.fx.experimental.symbolic_shapes, this seems fine.

Reply:
done